src/Nethermind/Nethermind.Sockets/IpcSocketMessageStream.cs

```diff
@@ -33,9 +35,18 @@ public class IpcSocketMessageStream(Socket socket) : NetworkStream(socket), IMes
             catch { }
         }
 
-        int read = _bufferedDataLength + await Socket.ReceiveAsync(buffer[_bufferedDataLength..], SocketFlags.None);
+        int delimiter = ((IList<byte>)buffer[.._bufferedDataLength]).IndexOf(Delimiter);
```

Comment:
At the end of the day I think they fall back to the same implementation:

```suggestion
        int delimiter = buffer[.._bufferedDataLength].AsSpan().IndexOf(Delimiter);
```